server/src/main/java/org/apache/druid/server/metrics/DruidMonitorSchedulerConfig.java

```diff
@@ -28,9 +29,17 @@
  */
 public class DruidMonitorSchedulerConfig extends MonitorSchedulerConfig
 {
+  @JsonProperty
+  private String schedulerClassName = ClockDriftSafeMonitorScheduler.class.getName();
```

Comment:
It was changed in https://github.com/apache/druid/pull/10448, not in this PR. This PR is just to make it configurable because I'm not sure how stable it is. As noted in https://github.com/apache/druid/pull/10448#issuecomment-756367474, CronScheduler seems to have a not-bad test coverage and worked well in my testing.

Reply:
I guess it will get tested in the next RC then.

Reply:
Yes, I have done some testing before and will do more.

Reply:
My 2¢: the best plan is to default to the old one, and then in the future (after some people have enabled the new one in production) we should switch to the new one, and remove the old one and remove the config entirely.

Rationale:

The new scheduler is designed to eliminate potential clock drift for monitors. This reward is real but is pretty small impact. I don't expect anything bad will happen if the schedule drifts a bit. The main risk of the new scheduler, I suppose, is that there's some case where it goes haywire, and either locks up completely or fires much more often than it should. I'm not sure how likely this is, but it's (a) hard to test for, (b) quite bad if it happens.

So, because the potential reward has a small impact, and the potential risk has a large impact, I think it's best to default to the old scheduler for another release or so. Just until such time as people have been able to do long-running tests in production and have found that there are no issues.

At any rate, it's good that this is undocumented, since it's an inside-baseball sort of config that we would only want to exist for a few releases.

Reply:
By the way, if anyone has been running the patch in production for a while already, now would be a good time to speak up. If we have already built up a good amount of confidence then I think it makes sense to default to the new one.

Reply:
I've also thought about this a bunch and have changed my opinion on whether or not we should change the scheduler to a new dependency by default a few times.

While changing to use the CronScheduler might fix a bug, it isn't clear whether any users have run into this in the field. I thought about documenting why a user would want to change the scheduler to CronScheduler instead of the older implementation, and I couldn't think of a good user facing reason to do so. So if we set the default to the old implementation, I don't think anyone would test it in production, so it would continue to live as dead code, and we'll have the same dilemma in the next release or 2 when we ask whether or not this has been run in production.

Setting the default to the older implementation reduces the impact of any bug that might show up in long running tests (even though this library was specifically built to fix issues found with long running processes). The drawback here is finding a reason for some users to try this in production so that we can sunset the feature flag in a release or 2.

Writing out this comment, I now think the more cautious approach - keeping the default the same - is better as it's hard to articulate the benefit for switching the scheduling and taking on the risk associated with changing the older behavior.

Reply:
> So, because the potential reward has a small impact, and the potential risk has a large impact, I think it's best to default to the old scheduler for another release or so. Just until such time as people have been able to do long-running tests in production and have found that there are no issues.

This makes sense to me. I think we can do more extensive testing by ourselves instead of rushing to change the default.

Reply:
Changed the default back to `BasicMonitorScheduler`.